File: RokidSDKDemoJava/app/src/main/java/com/rokid/mobile/sdk/demo/java/skill/SkillIndexFragment.java
package com.rokid.mobile.sdk.demo.java.skill;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.view.View;
import android.view.ViewGroup;

import com.rokid.mobile.sdk.demo.java.R;
import com.rokid.mobile.sdk.demo.java.adapter.BaseFragmentAdapter;
import com.rokid.mobile.sdk.demo.java.base.BaseFragment;
import com.rokid.mobile.sdk.demo.java.base.BaseFragmentPresenter;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * Created by tt on 2018/2/28.
 */

public class SkillIndexFragment extends BaseFragment {

    @BindView(R.id.skill_tab_layout)
    TabLayout skillTab;

    @BindView(R.id.skill_viewPager)
    ViewPager skillVp;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_skill_index;
    }

    @Override
    protected BaseFragmentPresenter initPresenter() {
        return null;
    }

    @Override
    protected void initVariables(View rootView, ViewGroup container, @Nullable Bundle savedInstanceState) {
        skillTab.setupWithViewPager(skillVp);
        List<BaseFragmentAdapter.Node> skillList = new ArrayList<>();
        skillList.add(new BaseFragmentAdapter.Node("闹钟", new SkillAlarmFragment()));
        skillList.add(new BaseFragmentAdapter.Node("提醒", new SkillRemindFragment()));
        skillList.add(new BaseFragmentAdapter.Node("智能家居", new SkillHomebaseFragment()));
        BaseFragmentAdapter mAdapter = new BaseFragmentAdapter(getFragmentManager(), skillList);

        skillVp.setAdapter(mAdapter);
    }

    @Override
    protected void initListeners() {
    }
}

File: RokidSDKDemoKotlin/app/src/main/kotlin/com/rokid/mobile/sdk/demo/skill/SkillIndexFragment.kt
package com.rokid.mobile.sdk.demo.skill

import com.rokid.mobile.sdk.demo.R
import com.rokid.mobile.sdk.demo.base.BaseFragment
import com.rokid.mobile.sdk.demo.base.BaseFragmentAdapter
import kotlinx.android.synthetic.main.skill_fragment_index.view.*

/**
 * Description: TODO
 * Author: Shper
 * Version: V0.1 2018/2/26
 */
class SkillIndexFragment : BaseFragment() {

    override fun layoutId(): Int = R.layout.skill_fragment_index

    override fun initViews() {
        rootView!!.skill_title.text = "技能"
        rootView!!.skill_subtitle.text = "SDK Skill Demo"

        rootView!!.skill_tab_layout.setupWithViewPager(rootView!!.skill_viewPager)
        rootView!!.skill_viewPager.adapter = BaseFragmentAdapter(activity!!.supportFragmentManager,
                SkillData.fragmentList)
    }

    override fun initListeners() {
    }

    object SkillData {
        val fragmentList = listOf(
                BaseFragmentAdapter.Companion.Node("闹钟", SkillAlarmFragment()),
                BaseFragmentAdapter.Companion.Node("提醒", SkillRemindFragment()),
                BaseFragmentAdapter.Companion.Node("媒体内容", SkillMediaFragment()),
                BaseFragmentAdapter.Companion.Node("智能家居", SkillHomebaseFragment()),
                BaseFragmentAdapter.Companion.Node("技能商店", SkillStoreFragment()),
                BaseFragmentAdapter.Companion.Node("酷狗登录", SkillKuGouFragment())
        )
    }

}